services/appflowy-collaborate/src/indexer/indexer_scheduler.rs

```diff
@@ -209,12 +220,17 @@ impl IndexerScheduler {
   pub fn index_encoded_collabs(
     &self,
     workspace_id: &str,
-    indexed_collabs: Vec<IndexedCollab>,
+    mut indexed_collabs: Vec<IndexedCollab>,
   ) -> Result<(), AppError> {
     if !self.index_enabled() {
       return Ok(());
     }
 
+    indexed_collabs.retain(|collab| self.is_indexing_enabled(&collab.collab_type));
+    if indexed_collabs.is_empty() {
+      return Ok(());
```

Comment:
The context in which this method is used means, that you can first deep copy 10 000 collabs (all of database row type => unindexable), so that you just drop them here.